src/Authentication/AuthenticatorConfigBase.cs

```diff
@@ -0,0 +1,52 @@
+﻿// <copyright file="AuthenticatorConfigBase.cs" company="ProjectBlackmagic">
+// RestfulClient
+// Copyright (c) ProjectBlackmagic. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+// Github: https://github.com/ProjectBlackmagic/RestfulClient.
+// </copyright>
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+
+namespace ProjectBlackmagic.RestfulClient.Authentication
+{
+    /// <summary>
+    /// Base class for authenticator configurations.
+    /// </summary>
+    public abstract class AuthenticatorConfigBase
+    {
+        /// <summary>
+        /// Gets certificate from local certificate store.
+        /// </summary>
+        /// <param name="certificateName">Certificate name</param>
+        /// <returns>Certificate from store</returns>
+        protected static X509Certificate2 GetCertificateFromStore(string certificateName)
+        {
+            using (var store = new X509Store(StoreName.My, StoreLocation.LocalMachine))
+            {
+                store.Open(OpenFlags.ReadOnly);
+                X509Certificate2Collection collection = store.Certificates.Find(X509FindType.FindBySubjectDistinguishedName, "CN=" + certificateName, true);
```

Comment:
How about exposing as well. Some people would need to use thumbprint instead (or something else).